homeassistant/components/mqtt/__init__.py

```diff
@@ -663,6 +665,25 @@ def _match_topic(subscription, topic):
     return reg.match(topic) is not None
 
 
+def _run_subscription_callback(hass, subscription: Subscription,
+                               msg) -> None:
+    """Prepare message payload and run the subscription callback."""
+    if subscription.callback is None:
```

Comment:
When can callback be `None` ?

Reply:
I added this to allow subscriptions without callbacks - i.e. allow `async_subscribe` to be called to just cause a MQTT client subscription to be made without having to supply an empty callback. Right now this is only used in tests, so it's probably best to remove this "feature".